sklearn/utils/extmath.py

```diff
@@ -599,8 +599,10 @@ def log_logistic(X, out=None):
     http://fa.bianp.net/blog/2013/numerical-optimizers-for-logistic-regression/
     """
     is_1d = X.ndim == 1
+    X = np.atleast_2d(X)
```

Comment:
The plan is to deprecate passing of 1d arrays. The first part of my PR was to make sure that all code works without check_array doing any reshaping. Now `check_array` is warning and reshaping anyways, but in later versions it will throw an error.